### PR TITLE
Add basic tests

### PR DESCRIPTION
**Pull request by kshitij12345**

### CODE REVIEW

File: tests/test_e2e.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        create_booster_summary,
+    )
+
+
+@pytest.mark.parametrize("log_importances", [True, False])
+@pytest.mark.parametrize("log_confusion_matrix", [True, False])
+@pytest.mark.parametrize("log_pickled_booster", [True, False])
+@pytest.mark.parametrize("log_trees", [True, False])
+@pytest.mark.parametrize("log_trees_as_dataframe", [True, False])
+def test_e2e(dataset, log_importances, log_confusion_matrix, log_pickled_booster, log_trees, log_trees_as_dataframe):
```
Comment (kshitij12345) [inline tests/test_e2e.py]: This generates 2^5 -> 32 tests. Is that ok?

Comment (twolodzko) [reply tests/test_e2e.py]: How long does it run? If the parameters are independent of each other and it's unlikely to change, then probably no need to check all the combinations.

File: .github/actions/e2e/action.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+     - name: Setup Graphviz
+       uses: ts-graphviz/setup-graphviz@v1
+       with:
+         ubuntu-skip-apt-update: true
+         macos-skip-brew-update: true
+
+     - name: Install scikit-learn
+       working-directory: ${{ inputs.working_directory }}
+       run: pip install scikit-learn
+       shell: bash
```
Comment (twolodzko) [inline .github/actions/e2e/action.yml]: This can be moved to "Install dependencies" step

File: tests/test_e2e.py (hunk 1 above)
Comment (kshitij12345) [reply tests/test_e2e.py]: It takes 2 to 9 mins on different platforms

image

Comment (kshitij12345) [reply tests/test_e2e.py]: They are independent (and I don't think that would change). Will reduce the combination.

**Comment by kshitij12345** [comment]
Gentle ping @twolodzko

File: .github/actions/e2e/action.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+   steps:
+     - name: Checkout
+       uses: actions/checkout@v2
+       with:
+         repository: neptune-ai/neptune-lightgbm
+         path: ${{ inputs.working_directory }}
+
+     - name: Install dependencies
+       working-directory: ${{ inputs.working_directory }}
+       run: pip install -e .[dev] && pip install scikit-learn
```
Comment (twolodzko) [inline .github/actions/e2e/action.yml:20]: No need to change it now, but FYI, you can also format it as multiline for improved readability:

```
run: |
   pip install -e .[dev]
   pip install scikit-learn
```

File: tests/test_e2e.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        log_importances,
+        log_confusion_matrix,
+        log_pickled_booster,
+        log_trees,
+        log_trees_as_dataframe,
+        base_namespace="",
+    )
+
+
+def test_e2e_using_namespace(dataset):
```
Comment (twolodzko) [inline tests/test_e2e.py:81]: Maybe we could also test it using a handler?

```
run = init_run()
handler = run["subdirectory"]
neptune_callback = NeptuneCallback(run=handler, base_namespace="training")
```

**Review by twolodzko — APPROVED** [review]
I added one comment on the possibility of creating an additional test for a handler. If possible, it'd be a good idea, if not, let's just merge it as-is.

File: .github/actions/e2e/action.yml (hunk 3 above)
Comment (kshitij12345) [reply .github/actions/e2e/action.yml:20]: Thanks!

File: tests/test_e2e.py (hunk 4 above)
Comment (kshitij12345) [reply tests/test_e2e.py:81]: Makes sense. Thank you!

**Pull request merged**